pkg/controller/adam/adam.go

```diff
@@ -11,26 +11,65 @@ import (
 	uuid "github.com/satori/go.uuid"
 	log "github.com/sirupsen/logrus"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"path"
+	"strconv"
+	"strings"
 	"time"
 )
 
 type Ctx struct {
-	dir         string
-	url         string
-	serverCA    string
-	insecureTLS bool
-	AdamRemote  bool
+	dir              string
+	url              string
+	serverCA         string
+	insecureTLS      bool
+	AdamRemote       bool
+	AdamRemoteRedis  bool   //use redis for obtain logs and info
+	AdamRedisUrlEden string //string with redis url for obtain logs and info
+}
+
+//parseRedisUrl try to use string from config to obtain redis url
+func parseRedisUrl(s string) (addr, password string, databaseID int, err error) {
+	URL, err := url.Parse(s)
+	if err != nil || URL.Scheme != "redis" {
+		return "", "", 0, err
+	}
+
+	if URL.Host != "" {
+		addr = URL.Host
+	} else {
+		addr = "localhost:6379"
```

Comment:
This magic still is here

Reply:
Yes, same behavior as in [adam](https://github.com/lf-edge/adam/blob/2a85416a46578d4b59dbfc9f95bfc2a24f99598a/pkg/driver/device_manager_redis.go#L104) we want to start. Work with consolidation of config planned in next PRs.